speechbrain/speechbrain
# Update full_inference test

Update `full_inference.csv` by using an annotation file based on LibriSpeech test-clean in SpeechBrain. 

It allows us to run the inference test much faster than before (about ~5 minutes for all entries in the CSV while in the past it could have taken more than 1.5 hours). 

Note: I set the WER performance value at 6.0. Should I change it to match exactly what we get with the pretrained models? 

## How to test it? 
Please run:
```bash
python -c 'from tests.utils.recipe_tests import run_recipe_tests; print("TEST FAILED!") if not(run_recipe_tests(filters_fields=["Task"], filters=[["full_inference"]], do_checks=True, run_opts="--device=cuda")) else print("TEST PASSED")'
``` 

The expected output is the following (if you already have download the pretrained models/datasets):
> (1/2) Running test for full_inference_row_2...
        download_file('https://www.openslr.org/resources/12/test-clean.tar.gz', 'tests/download/test-clean.tar.gz', unpack=True, dest_unpack='tests/download/LibriSpeech',write_permissions=True)
tests/download/test-clean.tar.gz exists. Skipping download
Extracting tests/download/test-clean.tar.gz to tests/download/LibriSpeech
        download_file('https://www.dropbox.com/sh/wy6ktfdxqt0dmou/AABY4ty3zy3xPef0zwRx949Oa?dl=1', 'tests/download/LibriSpeech_ASR_transformer.zip', unpack=True, dest_unpack='tests/download/LibriSpeech_ASR_transformer', write_permissions=True)
tests/download/LibriSpeech_ASR_transformer.zip exists. Skipping download
Extracting tests/download/LibriSpeech_ASR_transformer.zip to tests/download/LibriSpeech_ASR_transformer
                Running an Inference Test
        ... 37.34s
        ...checking files & performance...
(2/2) Running test for full_inference_row_3...
        download_file('https://www.openslr.org/resources/12/test-clean.tar.gz', 'tests/download/test-clean.tar.gz', unpack=True, dest_unpack='tests/download/LibriSpeech', write_permissions=True)
tests/download/test-clean.tar.gz exists. Skipping download
Extracting tests/download/test-clean.tar.gz to tests/download/LibriSpeech
        download_file('https://www.dropbox.com/sh/vs1os5mwh4necp0/AAAs7rMl4IO4N4s__BbVon6qa?dl=1', 'tests/download/LibriSpeech_ASR_seq2seq.zip', unpack=True, dest_unpack='tests/download/LibriSpeech_ASR_seq2seq', write_permissions=True)
tests/download/LibriSpeech_ASR_seq2seq.zip exists. Skipping download
Extracting tests/download/LibriSpeech_ASR_seq2seq.zip to tests/download/LibriSpeech_ASR_seq2seq
                Running an Inference Test
        ... 31.22s
        ...checking files & performance...
TEST PASSED
